docs/00-engineering/Php/drupal.md

```diff
@@ -0,0 +1,522 @@
+
+Most Drupal 8 and beyond projects at Bixal use some version of [Bixal drupal-project](https://github.com/Bixal/drupal-project). The readme file in your project should be the first thing you do when on-boarding on to a new project. It should contain answers to most questions about how to set up the project and where to get resources. If there are addtional questions contact the project team in Microsoft teams for help.
+
+### Drupal-specific Coding Standards
+
+* We follow [Drupal Coding Standards](https://www.drupal.org/docs/develop/standards) which includes:
+  * [PHP](https://drupal.org/coding-standards)
+  * [CSS](https://www.drupal.org/docs/develop/standards/css)
+  * [JS](https://www.drupal.org/docs/develop/standards/javascript) coding standards.
+  * [Writing Secure Code in Drupal 7](https://www.drupal.org/docs/7/security/writing-secure-code)
+  * [Writing Secure Code in Drupal 8](https://www.drupal.org/docs/8/security/writing-secure-code-for-drupal-8)
+* We ensure and test our coding standards using the built in automation in projects. Please check the readme.md of the project you are working on. Most projects will have a [Makefile](https://github.com/Bixal/drupal-project/blob/8.x/Makefile) or Python Script which makes this a near trivial effort. Certain projects or lead may enforce this effort using githook. All projects will test for this in Bitbucket pipelines, Gitlab, Github, CircleCi, etc.
+  * All module functions (even little helper functions) should have the module prefix.
+  * Database tables should have the module prefix.
+  * Variables should have the module prefix.
+  * Custom code, when possible, should be developed as a standalone library that a module can include and integrate with Drupal. This should mean using class namespaces and best OOP practices, and at that point we can use an alternative more flexible autoloading approach, like the one provided by composer.
+* We always try to use Drupal functions where they exist. This helps with upgrades (among other things).
+  * Use `drupal_get_path();` to create a path to a file in your module directory.
+  * Use `path_to_theme();` to create a path to a file in the current theme directory.
+* We use the theme system! When theming, don't hack module code when it is possible to use a theme function instead.
+  * The corollary: when coding modules keep all markup in theme functions!
+* We use render arrays and theme functions, so we can separate logic and presentation.
+* We try to keep modules that could be useful on another site completely generic from the start (a good technique is to develop in a fresh Drupal install rather then the actual site), and theme the site specific stuff in your site theme.
+
+
+
+## Git Commits
+
+* Put descriptive text in the commit log.
+* Begin the commit log test with the ticket number, e.g. GN-123 (JIRA).
+
+## Editor Configuration for Coding
+
+These are some example editor configuration values - each editor has it's own set:
+
```

Comment:
We should probably add that we have PHP Storm licenses available, if that is the developer's editor of choice. That can be helpful and you may avoid a few unnecessary questions here and there...

Reply:
agreed, can you go ahead and make that change? I know shortess got delayed with some project troubles and will probably require a few extra days to get to the rest of this!

Reply:
Will do. As well as the front-end items I mentioned before. Will try to get the PR in by EOD or tomorrow!